src/Hint/NewTypeDerivingStrategies.hs

```diff
@@ -0,0 +1,38 @@
+{-
+    Suggest using explicit deriving strategies for newtype instances
+
+<TEST>
+newtype Foo = Foo Int deriving (Show, Eq) -- newtype Foo = Foo Int deriving newtype (Show, Eq)
+newtype Foo = Foo { getFoo :: Int } deriving (Show, Eq) -- newtype Foo = Foo { getFoo :: Int } deriving newtype (Show, Eq)
+newtype Foo = Foo Int deriving stock Show
+</TEST>
+-}
+module Hint.NewTypeDerivingStrategies (newTypeDerivingStrategiesHint) where
+
+import Hint.Type
+
+newTypeDerivingStrategiesHint :: DeclHint
+newTypeDerivingStrategiesHint _ _ = newTypeDerivingStrategiesHintDecl
+
+newTypeDerivingStrategiesHintDecl :: Decl_ -> [Idea]
+newTypeDerivingStrategiesHintDecl x
+    | Just new <- newtypeDecl x
+    = [suggestN "Use DerivingStrategies" x new
+      ]
+newTypeDerivingStrategiesHintDecl _ = []
+
+newtypeDecl :: Decl_ -> Maybe Decl_
+newtypeDecl (DataDecl x1 x2@(NewType _) x3 x4 x5 x6)
+    | any hasNoStrategy x6 = Just $ DataDecl x1 x2 x3 x4 x5 (withNewtype <$> x6)
+
```

Comment:
Please delete this blank line - otherwise it starts to look like a partial pattern match.

Reply:
indeed, good catch